### PR TITLE
[Fix] Ignore atime and mtime if they are invalid

### PR DESCRIPTION
**Pull request by piranna**
Fixes issue #113

### CODE REVIEW

**Comment by phated** [comment]
@piranna the callback isn't called if we have mtime but it is invalid.  also, can you add a test or two for this?  thanks!

**Comment by piranna** [comment]
I've just added the tests.

**Comment by phated** [comment]
@piranna thanks. is there a way you can think of to test the file didn't receive an invalid timestamp? Maybe check it is within some range of Date.now()?  I'm not too sure here.  It just seems like testing we didn't change the vinyl object is the wrong thing to be testing.

File: lib/utimes.js [hunk 1]
```diff
@@ -2,11 +2,20 @@
...
+// http://stackoverflow.com/a/10589791/586382
+function validDate(date) {
+  return date instanceof Date && !isNaN(date.valueOf());
+}
+
 function utimes(writePath, stat, cb) {
-  if (stat.mtime) {
+  var mtime = stat.mtime;
+
+  if (validDate(mtime)) {
```
Comment (contra) [inline lib/utimes.js]: `if (validDate(mtime) && validDate(atime)) {

**Comment by contra** [comment]
We should land this ASAP - the latest version on npm (which is being used by 4.0) is outputting broken files and people are pretty upset about that.

**Comment by phated** [comment]
@contra we need a better way to test this.  feedback welcome

**Comment by phated** [comment]
@piranna can you please fix the tests ASAP so we can get this published?  Thanks

**Comment by phated** [comment]
I believe I fixed the tests to verify that they aren't setting invalid values over on the [invalid-utimes branch](https://github.com/gulpjs/vinyl-fs/tree/invalid-utimes)

**Comment by phated** [comment]
Deleted branch and merged to master.  Waiting for travis to run and bumping a patch release.

**Comment by phated** [comment]
Published as 2.2.1

**Comment by piranna** [comment]
Sorry for the delay, I've been out due to holydays, that's why I tried to had it ready the last friday. I'm happy this could be fixed.

**Pull request merged**